src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -79,6 +66,11 @@ internal sealed class MeterProviderSdk : MeterProvider
                 }
             }
 
+            if (this.reader is CompositeMetricReader)
+            {
+                this.compositeMetricReader = this.reader as CompositeMetricReader;
```

Comment:
nit: You don't really need the `is` check. I think you could just execute this `as` line and it ends up `null` if it isn't a `CompositeMetricReader`. Or...

```csharp
if (this.reader is CompositeMetricReader compositeMetricReader)
   this.compositeMetricReader = compositeMetricReader;
```